articles/openshift/howto-deploy-java-eap-app.md

```diff
@@ -0,0 +1,384 @@
+
+Before deploying the application, let's creating the expected secret that will hold specific application configuration. The Helm Chart will get the database user, password and name from the `mssqlserver-secret`secret object created before, and the driver version, the datasource JNDI name and the cluster password from the following secret:
+
+1. Execute the following to create the application config map:
+
+    ```bash
+    todo-list (bootable-jar-openshift) $ oc create secret generic todo-list-secret \
+    --from-literal app-driver-version=7.4.1.jre11 \
+    --from-literal app-ds-jndi=java:/comp/env/jdbc/mssqlds \
+    --from-literal app-cluster-password=mut2UTG6gDwNDcVW
+    ```
+
+    > [!NOTE]
+    > You decide which cluster password you want to use, the pods that want to join to your cluster need such password, that prevents your JBoss EAP cluster is not messed up with any pod that are not under your control and ask to join.
+
+    > [!NOTE]
+    > You could have noticed from the above config map that we are not supplying the database Hostname and Port. That's not necessary. If you take a closer look at the application Helm Chart file, you will see that the database Hostname and Port are passed by using the following notations \$(MSSQLSERVER_SERVICE_HOST) and \$(MSSQLSERVER_SERVICE_PORT). This is a standard OpenShift notation that will make the application variables (MSSQLSERVER_HOST, MSSQLSERVER_PORT) get assigned to the values of pod environment variables (MSSQLSERVER_SERVICE_HOST, MSSQLSERVER_SERVICE_PORT) available at runtime, which are passed by OpenShift when the pod is launched. These variables are available to any pod because we have created a Service to expose the SQL server in the previous steps.
+
+2. Open the OpenShift console and navigate to the developer view (in the `</> Developer` perspective in the left hand menu)
+
+    ![OpenShift console developer view](./media/howto-deploy-java-eap-app/console_developer_view.png)
+
+3. Once you are in the `</> Developer` perspective, ensure you have selected the `eap-demo` project at the projects combo box.
+
+    ![OpenShift console project combo box](./media/howto-deploy-java-eap-app/console_project_combo_box.png)
+
+4. Click on `+Add`, then Helm Chart. You will arrive at the Helm Chart catalog available on your ARO cluster. Write `eap` on the filter input box to filter all the Helm Charts and get the EAP ones. At this stage, you should see two options:
+
+    ![OpenShift console EAP Helm Charts](./media/howto-deploy-java-eap-app/console_eap_helm_charts.png)
+
+5. Since our application uses Microprofile capabilities, you have to select for this demo the one named EAP Xp (at the time of this writing, the exact version is EAP Xp 1.0.0). The `Xp` stands for Expansion Pack. With the JBoss Enterprise Application Platform expansion pack, developers can use Eclipse MicroProfile application programming interfaces (APIs) to build and deploy microservices-based applications.
```

Comment:
The XP casing is incorrect here: `s/EAP Xp/EAP XP/g` and the current version is EAP XP 3.0 (https://access.redhat.com/articles/2026253#CompatibleEAPXPVersions74)

Reply:
I think there is some incorrect casing on the OpenShift Helm Chart from your screen shot and this should be reported elsewhere and updated. But there are changes that should be made in this document too. According to your screen shot the version is "Eap Xp3 v1.0.0" (please note that there is the presence of 3 in that version)

Reply:
I have a vague recollection that the casing used in the console UI is out of our control. @yersan , if you don't already know if I'm right about that, Jeff would know.

Reply:
Jeff confirmed it is out of our control, there is the same issue with WildFly, which is rendered as Wildfly. OpenShift decides how it gets rendered.